src/miner.cpp

```diff
@@ -706,24 +702,40 @@ class MinerAddressScript : public CReserveScript
     void KeepScript() {}
 };
 
+std::optional<MinerAddress> ExtractMinerAddress::operator()(const CKeyID &keyID) const {
+    boost::shared_ptr<MinerAddressScript> mAddr(new MinerAddressScript());
+    mAddr->reserveScript = CScript() << OP_DUP << OP_HASH160 << ToByteVector(keyID) << OP_EQUALVERIFY << OP_CHECKSIG;
+    return mAddr;
+}
+std::optional<MinerAddress> ExtractMinerAddress::operator()(const CScriptID &addr) const {
+    return std::nullopt;
+}
+std::optional<MinerAddress> ExtractMinerAddress::operator()(const libzcash::SproutPaymentAddress &addr) const {
+    return std::nullopt;
+}
+std::optional<MinerAddress> ExtractMinerAddress::operator()(const libzcash::SaplingPaymentAddress &addr) const {
+    return addr;
+}
+std::optional<MinerAddress> ExtractMinerAddress::operator()(const libzcash::UnifiedAddress &addr) const {
+    for (const auto& receiver: addr) {
+        if (std::holds_alternative<libzcash::SaplingPaymentAddress>(receiver)) {
+            return std::get<libzcash::SaplingPaymentAddress>(receiver);
+        }
+    }
+    return std::nullopt;
+}
+
+
 void GetMinerAddress(MinerAddress &minerAddress)
```

Comment:
Is it intentional that this doesn't necessarily set `minerAddress`? That seems like a potential hazard that deserves a comment.

Reply:
This isn't a change in behavior, but I agree that this behavior is hazardous. If we change this though, I think we should change it to take no arguments and to return `std::optional<MinerAddress>`